enhancements/microshift/multus-cni-for-microshift.md

```diff
@@ -0,0 +1,401 @@
+Multus plugin.
+
+Even though these are limitation of the Multus itself, they should be documented.
+
+## Design Details
+
+### Open Questions [optional]
+
+1. Clean up of Multus artifacts on disk during system rollback or when `microshift-cleanup-data` is executed:
+   Should we invest in creating a lightweight plugin architecture to avoid adding cleanup of Multus
+   to MicroShift's core cleanup script? This architecture could be based on existing
+   [kubectl plugin](https://kubernetes.io/docs/tasks/extend-kubectl/kubectl-plugins/) architecture
+   where plugins start with `kubectl-`, so we could use `microshift-cleanup-plugin-` prefix.
+
+### Test Plan
+
+Multus CNI for MicroShift will be tested using existing test harness. New test suite will be created
+with simple test for each CNI we want to declare as supported.
+
+Starting with `bridge` test shall: install Multus, create NetworkAttachmentDefinition and deploy
+a Pod that should have additional interface attached. After verifying that the interface is present,
+test will try to access Pod's application using the bridge interface on the host to make sure there
+is a connectivity.
+
+Tests for other networking plugins will be designed and implemented when plugins are planned for support.
+
+### Graduation Criteria
+
+Multus CNI for MicroShift is targeted to be GA next release.
+
+### Upgrade / Downgrade Strategy
+
+Because both Multus and MicroShift RPMs will be build from the same spec file, they will share version
+and it is expected that they are updated together following MicroShift upgrade rules depending on
+type of operating system (ostree-based or regular RPM).
+
+Considering only the manifests, we know that on each start MicroShift will apply manifests forcefully
+overwriting any differences. However, MicroShift does not have any uninstall capabilities.
+If manifests ever change, for example some ConfigMap is renamed, then these old parts will
+keep existing in the etcd. This could have undesirable consequences of having two Multus DaemonSets
+if name or namespace of original DaemonSet changes. To make potential the transition to thick Multus
+smooth, we should not deviate from already existing resource names present in OpenShift's Multus
+manifests. This problem is not Multus specific - it is how MicroShift works and it is not part of this
+enhancement addressing this shortcoming.
+
+Because of the way Multus works (checks Annotations when needed, executed when networking must be
+set up, does not keep its own database beside cache) we can consider it to be mostly stateless.
+When Multus CNI binary is executed by kubelet/CRI-O it will setup Pod's network according to the
+configs and Annotations (not a database with "expected states"), therefore updating Multus does not
+require migration strategies.
+
+If schema of NetworkAttachmentDefinitions CRD changes, instead of deploying a mutating webhook,
+we can just suggest users to update their CRs.
```

Comment:
Yes, preparing new ostree commit is rather conscious decision so I think it'd be better to prepare new file up front instead of using webhook that uses resources idling 99.999% time. 
I'll expand that